### ISSUE TITLE
`jsdoc/no-undefined-types` throws error for local class properties

### ISSUE THREAD

**Issue by PartyWumpus** [gajus/eslint-plugin-jsdoc#1760]
## Expected behavior

I should be able to reference a property on the current class I'm documenting without using the class' full name. (as in `{@link property_a}` instead of `{@link MyClassName.property_a}`. (Typescript understands the annotation)

## Actual behavior

Raises an error `jsdoc/no-undefined-types: The type 'property' is undefined` on the line where the doc is.

## ESLint Config

```js
'jsdoc/no-undefined-types': 'error'
```

## ESLint sample

```typescript
class MyClass {
	public property: boolean

	/** xyz {@link property} */
	public other_property_or_func: boolean
}
```

or (arguably a different issue idk)

```typescript
/** xyz {@link property} */
class MyClass {
	public property: boolean
}
```

## Environment

- Node version: v24.19.0
- ESLint version v10.9.1
- `eslint-plugin-jsdoc` version: 64.3.3

**Comment by PartyWumpus** [comment]
wow that was fast. thanks!

**Comment by PartyWumpus** [comment]
this doesn't fix the
```
/** xyz {@link property} */
class MyClass {
	public property: boolean
}
```
case

should i open a seperate issue?

**Comment by brettz9** [comment]
yes
